server/circleci.go

```diff
@@ -141,40 +147,33 @@ type PipelineWorkflowResponse struct {
 
 func (s *Server) waitForWorkflowId(ctx context.Context, id string, workflowName string) (string, error) {
 	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
 	for {
 		select {
 		case <-ctx.Done():
-			ticker.Stop()
 			return "", errors.New("timed out trying to fetch workflow")
 		case <-ticker.C:
-			req, err := http.NewRequest("GET", "https://circleci.com/api/v2/pipeline/"+id+"/workflow", nil)
+			req, err := http.NewRequestWithContext(ctx, "GET", "https://circleci.com/api/v2/pipeline/"+id+"/workflow", nil)
 			if err != nil {
 				return "", err
 			}
 			req.SetBasicAuth(os.ExpandEnv(s.Config.CircleCIToken), "")
-
 			resp, err := http.DefaultClient.Do(req)
 			if err != nil {
 				return "", err
 			}
-			b, err := ioutil.ReadAll(resp.Body)
-			err = resp.Body.Close()
-			if err != nil {
-				return "", err
-			}
-
 			if resp.StatusCode != http.StatusOK {
 				continue
 			}
-
-			triggeredR := PipelineWorkflowResponse{}
-			err = json.Unmarshal(b, &triggeredR)
+			r := PipelineWorkflowResponse{}
+			defer resp.Body.Close()
```

Comment:
This is a bug. The defer call should be made immediately after the err check. Right now, the body won't be closed if we get a 500 response.